photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LinearRegressionOptimizationProblemTest.scala

```diff
@@ -14,24 +14,62 @@
  */
 package com.linkedin.photon.ml.optimization
 
-import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
-import com.linkedin.photon.ml.model.Coefficients
-import com.linkedin.photon.ml.normalization.NormalizationContext
-import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
-import com.linkedin.photon.ml.test.CommonTestUtils
+import java.util.Random
 
+import breeze.linalg.{DenseVector, Vector}
 import org.apache.spark.rdd.RDD
 import org.mockito.Mockito._
 import org.testng.Assert._
-import org.testng.annotations.Test
+import org.testng.annotations.{DataProvider, Test}
 
-class LinearRegressionOptimizationProblemTest {
-  import LinearRegressionOptimizationProblemTest._
+import com.linkedin.photon.ml.constants.MathConst
+import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
+import com.linkedin.photon.ml.model.Coefficients
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
+import com.linkedin.photon.ml.test.{CommonTestUtils, SparkTestUtils}
+
+class LinearRegressionOptimizationProblemTest extends SparkTestUtils {
   import CommonTestUtils._
+  import LinearRegressionOptimizationProblemTest._
+
+  def generateUnweightedBenignLocalDataSet: List[LabeledPoint] = {
+    drawSampleFromNumericallyBenignDenseFeaturesForLinearRegressionLocal(DATA_RANDOM_SEED, TRAINING_SAMPLES, DIMENSIONS)
+      .map { obj =>
+        val (label, features) = obj
+        assertEquals(features.length, DIMENSIONS, "Samples should have expected lengths")
+
+        new LabeledPoint(label, features)
+      }
+      .toList
+  }
+
+  def generateWeightedBenignLocalDataSet: List[LabeledPoint] = {
+    val r: Random = new Random(WEIGHT_RANDOM_SEED)
+
+    drawSampleFromNumericallyBenignDenseFeaturesForLinearRegressionLocal(DATA_RANDOM_SEED, TRAINING_SAMPLES, DIMENSIONS)
+      .map { obj =>
+        val (label, features) = obj
+        val offset = 0D
+        val weight = r.nextDouble() * WEIGHT_MAX
+        assertEquals(features.length, DIMENSIONS, "Samples should have expected lengths")
+
+        new LabeledPoint(label, features, offset, weight)
+      }
+      .toList
+  }
+
+  @DataProvider(parallel = true)
+  def getDataAndWeights: Array[Array[Object]] = {
+    val weightsToTest = Array(0.1, 1.0, 10.0, 100.0)
+    val dataSet = generateWeightedBenignLocalDataSet
+
+    weightsToTest.map( Array(_, dataSet).asInstanceOf[Array[Object]] )
+  }
 
   @Test
   def testUpdateObjective(): Unit = {
-    val problem = createProblem
+    val problem = createProblem()
```

Comment:
As per our talk offline: functions that have parameters must have the parentheses, even if all default arguments are used. Scala compiler enforces it.